### PR TITLE
regols: init at 0.2.2

### PR DESCRIPTION
**Pull request by alias-dev**
###### Description of changes

Adds the [OPA Rego language server](https://github.com/kitagry/regols), and myself as maintainer.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/language-servers/regols/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  version = "0.2.2";
+
+  src = fetchFromGitHub {
+    owner = "kitagry";
+    repo = "regols";
+    rev = "v${version}";
+    sha256 = "sha256-2ZwmIlv3kJ26p15t7NvB9sX2GO+B3ypeNl50b7XA0Iw=";
+  };
+
+  vendorSha256 = "sha256-N6gtkZSNLXz3B961grM3xHzm7x4/kzcLkDOgiFLGp8U=";
```
Comment (doronbehar) [inline pkgs/development/tools/language-servers/regols/default.nix:14]: Nit:

```suggestion
  vendorHash = "sha256-N6gtkZSNLXz3B961grM3xHzm7x4/kzcLkDOgiFLGp8U=";
```

File: pkgs/development/tools/language-servers/regols/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+buildGoModule rec {
+  pname = "regols";
+  version = "0.2.2";
+
+  src = fetchFromGitHub {
+    owner = "kitagry";
+    repo = "regols";
+    rev = "v${version}";
+    sha256 = "sha256-2ZwmIlv3kJ26p15t7NvB9sX2GO+B3ypeNl50b7XA0Iw=";
```
Comment (doronbehar) [inline pkgs/development/tools/language-servers/regols/default.nix:11]:
```suggestion
    hash = "sha256-2ZwmIlv3kJ26p15t7NvB9sX2GO+B3ypeNl50b7XA0Iw=";
```

File: pkgs/development/tools/language-servers/regols/default.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  };
+
+  vendorSha256 = "sha256-N6gtkZSNLXz3B961grM3xHzm7x4/kzcLkDOgiFLGp8U=";
+
+  meta = with lib; {
+    description = "OPA Rego language server";
+    homepage = "https://github.com/kitagry/regols/tree/main";
```
Comment (doronbehar) [inline pkgs/development/tools/language-servers/regols/default.nix:18]:
```suggestion
    homepage = "https://github.com/kitagry/regols";
```

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -17451,6 +17451,8 @@ with pkgs;
 
   nil = callPackage ../development/tools/language-servers/nil { };
 
+  regols = callPackage ../development/tools/language-servers/regols { };
```
Comment (doronbehar) [inline pkgs/top-level/all-packages.nix:17454]: There's a merge conflict in this file. Specifically for this case [RFC 140 was created](https://github.com/NixOS/rfcs/pull/140). You can switch to put files in `pkgs/by-name.` as explained in the RFC.

**Review by doronbehar — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix (hunk 4 above)
Comment (alias-dev) [reply pkgs/top-level/all-packages.nix]: Thanks @doronbehar, I've made the requested changes, and have rebuilt and tested the binary. Is this now looking OK?

**Review by doronbehar — APPROVED** [review]

**Pull request merged**